packages/gatsby-cli/src/index.js

```diff
@@ -24,10 +25,17 @@ if (!semver.satisfies(process.version, MIN_NODE_VERSION)) {
   )
 }
 
-process.on(`unhandledRejection`, error => {
+process.on(`unhandledRejection`, reason => {
   // This will exit the process in newer Node anyway so lets be consistent
   // across versions and crash
-  report.panic(`UNHANDLED REJECTION`, error)
+
+  // reason can be anything, it can be a message, an object, ANYTHING!
+  // we convert it to an error object so we don't crash on structured error validation
+  if (!(reason instanceof Error)) {
+    reason = new Error(util.format(reason))
```

Comment:
no idea how to improve this, so I'm happy to accept suggestions

Reply:
Yeah, seems pretty reasonable 👍